python/lsst/pipe/tasks/dataFrameActions/_actions.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+__all__ = ("SingleColumnAction", "MultiColumnAction", "CoordColumn", "MagColumn", "SumColumns", "AddColumn",
+           "DivideColumns", "SubtractColumns", "MultiplyColumns",)
+
+from typing import Iterable
+
+import numpy as np
+import pandas as pd
+from astropy import units
+
+from ..configurableActions import ConfigurableActionStructField, ConfigurableActionField
+from ._baseDataFrameActions import DataFrameAction
+from ._evalColumnExpression import makeColumnExpressionAction
+
+from lsst.pex.config import Field
+
+
+class SingleColumnAction(DataFrameAction):
+    column = Field(doc="Column to load for this action", dtype=str, optional=False)
+
+    @property
+    def columns(self) -> Iterable[str]:
+        return (self.column, )
+
+    def __call__(self, df, **kwargs):
+        return df[self.column]
+
+
+class MultiColumnAction(DataFrameAction):
+    actions = ConfigurableActionStructField(doc="Configurable actions to use in a joint action")
+
+    @property
+    def columns(self) -> Iterable[str]:
+        yield from (column for action in self.actions for column in action.columns)
+
+
+class CoordColumn(SingleColumnAction):
+    inRadians = Field(doc="Return the column in radians if true", default=True, dtype=bool)
+
+    def __call__(self, df):
+        col = super().__call__(df)
+        return col * 180 / np.pi if self.inRadians else col
+
+
+class MagColumn(SingleColumnAction):
+    coadd_zeropoint = Field(doc="Magnitude zero point", dtype=float, default=27)
+
+    def __call__(self, df: pd.DataFrame, **kwargs):
+        if not (fluxMag0 := kwargs.get('fluxMag0')):
+            fluxMag0 = 1/np.power(10, -0.4*self.coadd_zeropoint)
+
+        with np.warnings.catch_warnings():
+            np.warnings.filterwarnings('ignore', r'invalid value encountered')
+            np.warnings.filterwarnings('ignore', r'divide by zero')
+            return -2.5 * np.log10(df[self.column] / fluxMag0)
```

Comment:
As the plotting code runs on the object tables do you think it is worth making MagColumn by default operate on nanojanskies? The plotting code is currently the main user of these functions but I don't know if it will be forever.

Reply:
I changed it to MagColumnDN to leave room for something from Janskeys or something